File: app/src/main/java/com/stupidtree/cloudliter/ui/chat/detail/PopUpImageMessageDetail.java
package com.stupidtree.cloudliter.ui.chat.detail;

import android.annotation.SuppressLint;
import android.view.View;
import android.widget.TextView;

import com.stupidtree.cloudliter.R;
import com.stupidtree.cloudliter.data.model.ChatMessage;
import com.stupidtree.cloudliter.ui.widgets.TransparentBottomSheetDialog;

import org.jetbrains.annotations.NotNull;

import java.text.DecimalFormat;
import java.util.HashMap;

import butterknife.BindView;

/**
 * 圆角的文本框底部弹窗
 */
@SuppressLint("NonConstantResourceId")
public class PopUpImageMessageDetail extends TransparentBottomSheetDialog {
    /**
     * View绑定区
     */
    @BindView(R.id.neutral)
    TextView neutral;

    @BindView(R.id.drawing)
    TextView drawing;

    @BindView(R.id.hentai)
    TextView hentai;

    @BindView(R.id.porn)
    TextView porn;

    @BindView(R.id.sexy)
    TextView sexy;

    /**
     * 不得已放在UI里的数据
     */
    @NotNull
    ChatMessage chatMessage;



    public PopUpImageMessageDetail setChatMessage(@NotNull ChatMessage chatMessage) {
        this.chatMessage = chatMessage;
        return this;
    }


    @Override
    protected int getLayoutId() {
        return R.layout.activity_chat_popup_image_message_detail;
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    protected void initViews(View v) {
        DecimalFormat df = new DecimalFormat("#.####");
        HashMap<String,Float> map = chatMessage.getExtraAsImageAnalyse();
        neutral.setText(df.format(map.get("Neutral")));
        drawing.setText(df.format(map.get("Drawing")));
        porn.setText(df.format(map.get("Porn")));
        hentai.setText(df.format(map.get("Hentai")));
        sexy.setText(df.format(map.get("Sexy")));

    }



}

File: app/src/main/java/com/stupidtree/cloudliter/ui/chat/detail/PopUpImageMessageDetail.kt
package com.stupidtree.cloudliter.ui.chat.detail

import android.annotation.SuppressLint
import android.view.View
import com.stupidtree.cloudliter.R
import com.stupidtree.cloudliter.data.model.ImageEntity
import com.stupidtree.cloudliter.databinding.ActivityChatPopupImageMessageDetailBinding
import com.stupidtree.style.widgets.TransparentBottomSheetDialog
import java.text.DecimalFormat

/**
 * 圆角的文本框底部弹窗
 */
@SuppressLint("NonConstantResourceId")
class PopUpImageMessageDetail : TransparentBottomSheetDialog<ActivityChatPopupImageMessageDetailBinding>() {


    /**
     * 不得已放在UI里的数据
     */
    private lateinit var imageEntity: ImageEntity
    fun setChatMessage(imageEntity:ImageEntity): PopUpImageMessageDetail {
        this.imageEntity = imageEntity
        return this
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_chat_popup_image_message_detail
    }

    override fun initViews(v: View) {
        val df = DecimalFormat("#.####")
        val map = imageEntity.getExtraAsImageAnalyse()
        binding.neutral.text = df.format(map["Neutral"])
        binding.drawing.text = df.format(map["Drawing"])
        binding.porn.text = df.format(map["Porn"])
        binding.hentai.text = df.format(map["Hentai"])
        binding.sexy.text = df.format(map["Sexy"])
    }

    override fun initViewBinding(v: View): ActivityChatPopupImageMessageDetailBinding {
        return ActivityChatPopupImageMessageDetailBinding.bind(v)
    }
}